src/main/java/org/jabref/gui/JabRefFrame.java

```diff
@@ -1422,4 +1463,22 @@ private void updateTexts(UndoChangeEvent event) {
             */
         }
     }
+
+    private void copyGroupTreeNode(LibraryTab libraryTabCasted, Optional<GroupTreeNode> parent, Optional<GroupTreeNode> groupTreeNodeToCopy, List<BibEntry> allEntries) {
+        // add groupTreeNodeToCopy to the parent-- in the first run that will the source/main GroupTreeNode
+        GroupTreeNode copiedNode = parent.get().addSubgroup(groupTreeNodeToCopy.get().copyNode().getGroup());
```

Comment:
Passing Optional to methods is generally a bad practice. Plus, I don't see how they're useful in this context, because *parent* and *groupTreeNodeToCopy* are being called without checking whether they exist or not.

Reply:
Noted, thanks.